bots/nymphora/nymphora_main.lua

```diff
@@ -0,0 +1,613 @@
+	self:oncombateventOld(EventData)
+
+	local nAddBonus = 0
+	
+	if EventData.Type == "Ability" then
+		-- No use bonuses. Add up bonuses to not drop aggro as skills are no longer 'up'
+		if EventData.InflictorName == "Ability_Fairy1" then
+			nAddBonus = nAddBonus + object.nHealUp
+			object.nGraveyardUseTime = EventData.TimeStamp
+		elseif EventData.InflictorName == "Ability_Fairy3" then
+			nAddBonus = nAddBonus + object.nHealUp
+		end
+	elseif EventData.Type == "Item" then
+		if EventData.SourceUnit == core.unitSelf:GetUniqueID() then
+			if EventData.InflictorName == "Item_Morph" then
+				nAddBonus = nAddBonus + object.nSheepstickUp
+			elseif EventData.InflictorName == "Item_FrostfieldPlate" then
+				nAddBonus = nAddBonus + object.nFrostfieldUp
+			end
+		end
+	end
+	
+	if nAddBonus > 0 then
+		--decay before we add
+		core.DecayBonus(self)
+	
+		core.nHarassBonus = core.nHarassBonus + nAddBonus
+	end
+end
+object.oncombateventOld = object.oncombatevent
+object.oncombatevent 	= object.oncombateventOverride
+
+function object:SkillBuild()
+	local unitSelf = self.core.unitSelf
+
+	if skills.heal == nil then
+		skills.heal		= unitSelf:GetAbility(0)
+		skills.mana		= unitSelf:GetAbility(1)
+		skills.stun		= unitSelf:GetAbility(2)
+		skills.teleport	= unitSelf:GetAbility(3)
+		skills.recall	= unitSelf:GetAbility(5)
+	end
+
+	if unitSelf:GetAbilityPointsAvailable() <= 0 then
+		return
+	end
+
+	for i = 0, unitSelf:GetAbilityPointsAvailable(), 1 do
+		bAbilityLeveled = false
+		if skills.teleport:CanLevelUp() then
+			skills.teleport:LevelUp()
+		else
+			if (self.nHealNeeded > 1 and skills.heal:CanLevelUp()) or (self.nManaNeeded > 1 and skills.mana:CanLevelUp()) or
+				(not skills.stun:CanLevelUp() and (skills.heal:CanLevelUp() or skills.mana:CanLevelUp())) then
+				if self.nHealNeeded > self.nManaNeeded and skills.heal:CanLevelUp() then
+					skills.heal:LevelUp()
+					bAbilityLeveled = true
+					self.nHealNeeded = self.nHealNeeded - 1
+					--self.nHealNeeded = 0
+				else
+					skills.mana:LevelUp()
+					bAbilityLeveled = true
+					self.nManaNeeded = self.nManaNeeded - 1
+					--self.nManaNeeded = 0
+				end
+			elseif skills.stun:CanLevelUp() then
+				skills.stun:LevelUp()
+				bAbilityLeveled = true
+			end
+		end
+		if not bAbilityLeveled then
+			unitSelf:GetAbility(4):LevelUp()
+		end
+	end
+end
+
+function useHeal(botBrain, pos)
+	object.nHealLastCastTime = HoN.GetGameTime()
+	object.vecHealPos = pos
+	return core.OrderAbilityPosition(botBrain, skills.heal, pos)
+end
+
+behaviorLib.SupportBehavior = {}
+
+-- Base 10
+-- 0.5 for every missing % of hp or mana
+-- Max 60 at 0% of hp or mana
+function behaviorLib.SupportUtility(botBrain)
+	local unitSelf = core.unitSelf
+
+	local localUnits = core.localUnits
+	local allyHeroes = core.CopyTable(localUnits.AllyHeroes)
+	allyHeroes[unitSelf:GetUniqueID()] = unitSelf
+
+	local sType = ""
+	local nUtility = 0
+	local unitTarget = {}
+
+	local canGiveMana = skills.mana:CanActivate()

```

Comment:
bCanGiveMana, search for 'local' and just check through everything at the end of the other fixes